File: backend/Apprilium.TripPlanner/Apprilium.TripPlanner.Domain/Common/PagingParameter.cs


namespace Apprilium.TripPlanner.Domain.Common
{
    public class PagingParameter : ResourceParameters
    {
        private int? _pageSize { get; set; }
        public int PageNumber { get; set; } = 1;
        public string SortColumn { get; set; }
        public string SortOrder { get; set; } //"DESC" vs "ASC"
        public int Table { get; set; }
        public int Dropdown { get; set; }
        public bool All { get; set; }
        public int? PageSize
        {
            get => _pageSize;
            set => _pageSize = value != null && value <= 0 ? 10 : value;
        }
    }
}

File: frentend/Trip_States_Management/packages/common/src/models/PagingParameter.ts
export class PaginPatameter{

    private pageSize :number=25;
    public  pageNumber :number=1;
    public sortColumn :string|undefined;
    public sortOrder :string|undefined; //"DESC" vs "ASC"
    public table:number|undefined;
    public dropdown:number|undefined;
    public all:boolean;

    public constructor(pageSize: number, pageNumber:number,sortColumn:string|undefined,
        sortOrder :string|undefined,table:number|undefined,dropdown:number|undefined, all:boolean ){
             this.pageSize = pageSize != undefined && pageSize <=0 ? 10 : pageSize;
             this.pageNumber = pageNumber;
             this.sortColumn = sortColumn;
             this.sortOrder = sortOrder;
             this.table = table;
             this.all = all;
    }

  }